appserver/web/war-util/src/main/java/org/glassfish/web/loader/JarFileManager.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2023 Eclipse Foundation and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.web.loader;
+
+import com.sun.enterprise.util.io.FileUtils;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.System.Logger;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import static java.lang.System.Logger.Level.DEBUG;
+import static java.lang.System.Logger.Level.TRACE;
+import static java.lang.System.Logger.Level.WARNING;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.glassfish.web.loader.LogFacade.getString;
```

Comment:
Static import lines should be in top.

Reply:
> Static import lines should be in top.

This is not defined on GF yet, most classes have them closer to the impl (I prefer this), less at the top and some in between (which is the most ugly).

Reply:
It violates standard conventions and default rules of a lot of static code analyzers (sonar, checkstyle, etc).

Reply:
I believe that depends on configuration, there is no such global standard.

I use SonarQube and this is not seen as violation. I even tried to find such rule, but to my surprise I couldn't. I know it is supported in checkstyle and I too thought it is in SonarQube too.

Btw pretty much work to do with GF. I couldn't run it for a while, because Sonar analysis crashes for OOME now :-(
https://sonarcloud.io/project/overview?id=dmatej_glassfish

Reply:
Okay.

Reply:
To avoid future discussions and conflicts I started trivial survey ... at least it could be representative between friends of GlassFish :-)
Based on that I will probably introduce rule in Checkstyle and fix all classes.
https://twitter.com/dmatej/status/1614963057842688000